src/Mod/TechDraw/App/DrawRichAnno.cpp

```diff
@@ -65,24 +65,23 @@ DrawRichAnno::~DrawRichAnno()
 void DrawRichAnno::onChanged(const App::Property* prop)
 {
     if (!isRestoring()) {
-        if ((prop == &AnnoText) ||
-            (prop == &ShowFrame) ||
-            (prop == &MaxWidth) ) {
-            requestPaint();
-        }
+        return;
     }
+
+    if ((prop == &AnnoText) ||
+        (prop == &ShowFrame) ||
+        (prop == &MaxWidth) ) {
+        requestPaint();
+    }
+    
     DrawView::onChanged(prop);
 
 }
 
 short DrawRichAnno::mustExecute() const
 {
-    bool result = 0;
-    if (!isRestoring()) {
-        result =  (AnnoText.isTouched());
-    }
-    if (result) {
-        return result;
+    if (!isRestoring() && AnnoText.isTouched()) {
+        return true;

```

Comment:
A `short` function is is returning a `bool`.

Reply:
Yeah, it is meant to taking the previous code into consideration. But I can change it to `return 1` if needed.

Reply:
In general not confident of that. If there's a variable name that describes what "1" means that would be clearer. Otherwise you're only making things a little difficult.

Reply:
I'm not sure I am following you here. I can change it to `return 1` (instead of `return true`) if needed. I just refactored the code - I don't even know what the return value `1` means myself.
So I can only offer to change it to `return 1` (instead of `return true`)

Reply:
What I mean is that `return 1` doesn't make things clear. What does the value 1 mean? Instead, if there is an enum that says "must execute", "must not execute", "something is not clear" or such, then picture is clearer. However, this is coming from the core, so an change will have to happen there first. At this time I don't have an answer what to do. As such, you can go ahead and do `return 1`. This was just a comment.

Reply:
@WandererFan Let's change it to `return 1` then. I suggest that we do this in a separate PR afterwards, so I won't have to rebase every commit in all the 6 PRs already made.

Reply:
@benj5378 you could just make a single commit for all the files. As such it's OK to squash all these commits into a single one.